mozilla_schema_generator/glean_ping.py

```diff
@@ -13,20 +13,55 @@
 class GleanPing(GenericPing):
 
     schema_url = "https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/dev/schemas/glean/baseline/baseline.1.schema.json" # noqa E501
-    probes_url = "https://probeinfo.telemetry.mozilla.org/glean/{}/metrics"
+    probes_url_template = "https://probeinfo.telemetry.mozilla.org/glean/{}/metrics"
     repos_url = "https://probeinfo.telemetry.mozilla.org/glean/repositories"
 
-    default_probes_url = probes_url.format("glean")
+    default_dependencies = ['glean']
     default_pings = {"baseline", "events", "metrics"}
     ignore_pings = {"default", "glean_ping_info", "glean_client_info"}
 
     def __init__(self, repo):  # TODO: Make env-url optional
-        super().__init__(self.schema_url, self.schema_url, self.probes_url.format(repo))
+        self.repo = repo
+        super().__init__(
+            self.schema_url,
+            self.schema_url,
+            self.probes_url_template.format(repo)
+        )
+
+    def get_dependencies(self):
+        # Get all of the library dependencies for the application that
+        # are also known about in the repositories file.
+
+        # The dependencies are specified using library names, but we need to
+        # map those back to the name of the repository in the repository file.
+
+        dependencies = self._get_json(self.dependencies_url.format(self.repo))
+        dependency_library_names = list(dependencies.keys())
+
+        repos = self.get_repos()
+        repos_by_dependency_name = {}
+        for repo in repos:
+            for library_name in repo.get('library_names', []):
+                repos_by_dependency_name[library_name] = repo
+
+        dependencies = []
+        for name in dependency_library_names:
+            if name in repos_by_dependency_name:
+                dependencies.append(repos_by_dependency_name[name])
+
+        if not len(dependencies):
+            return self.default_dependencies
```

Comment:
Should we log something here as well?

Reply:
I'd log maybe only if a `dependencies_url` was defined on the repo, (i.e. it's an app, not a library), and no dependencies were found.  That's something that should be resolved.  Libraries will never have to define this, even if they have dependencies, since the app returns a flattened tree of *all* of its dependencies.

Reply:
Looking at this some more, I don't think this is really an error case.  If reading the git hash of an application *before* the dependencies artifact is being emitted (as is currently the case with Fenix), that's not an error -- we just fall back to "Glean only" as a dependency.

Reply:
I think that, in general, I'm fine with having the fewest possible log lines. What I'd really like out of this is to speed up diagnosis of potential bugs: instead of my comments above, even something as easy as "only glean dependencies for project X" might work. It tells us straight away what to expect from the data :)

Reply:
That makes sense.  I've updated this to just log the dependencies found (and whether it's just using the defaults).